### PR TITLE
smoketest for max6643 fan controller

### PR DESCRIPTION
**Pull request by nourishnew**

### CODE REVIEW

File: projects/smoketest_max6643/src/main.c [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_it_init();
+  gpio_init();
+  GpioAddress fanfail_pin_address = FANFAIL_PIN;
+  GpioAddress overtemp_pin_address = OVERTEMP_PIN;
+  GpioAddress fan_pin_address = FAN_PIN;
+  bool fan_speed_toggle = FAN_SPEED_HIGH;
```
Comment (copperium) [inline projects/smoketest_max6643/src/main.c:32]: You don't need to assign the #define'd macros to variables before using them, you can just use them directly.

File: projects/smoketest_max6643/src/main.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    .direction = GPIO_DIR_OUT,  //
+    .state = GPIO_STATE_LOW,    //
+  };
+  if (fan_speed_toggle) {
+    gpio_init_pin(&fan_pin_address, &gpio_high_settings);
+    gpio_set_state(&fan_pin_address, GPIO_STATE_HIGH);
+  } else {
+    gpio_init_pin(&fan_pin_address, &gpio_low_settings);
+    gpio_set_state(&fan_pin_address, GPIO_STATE_LOW);
+  }
```
Comment (copperium) [inline projects/smoketest_max6643/src/main.c:47]: `gpio_init_pin` already sets the state of the pin as set by the `state` field of `GpioSettings`, so you don't need the `gpio_set_state` calls.

File: projects/smoketest_max6643/src/main.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#include "max6643_fan_controller.h"
+#include "wait.h"
+
+#define FANFAIL_PIN \
+  { .port = GPIO_PORT_B, .pin = 4 }
+#define OVERTEMP_PIN \
+  { .port = GPIO_PORT_B, .pin = 5 }
+#define FAN_PIN \
+  { .port = GPIO_PORT_B, .pin = 3 }
+#define FAN_SPEED_HIGH true
```
Comment (copperium) [inline projects/smoketest_max6643/src/main.c:15]: This controls full speed, not "high speed", so `FAN_FULL_SPEED` might be a better name.

File: projects/smoketest_max6643/src/main.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
+#include <stdbool.h>
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "wait.h"
+
```
Comment (copperium) [inline projects/smoketest_max6643/src/main.c:8]: Generally we put a comment at the start of smoke tests explaining how to use it and what each of the configurable items (i.e. the defines) mean. This helps the hardware person using the smoke test to validate the board to understand how to use the smoke test.

**Review by copperium — CHANGES_REQUESTED** [review]

File: projects/smoketest_max6643/src/main.c [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_it_init();
+  gpio_init();
+  GpioAddress fanfail_pin_address = { .port = GPIO_PORT_B, .pin = 4 };
+  GpioAddress overtemp_pin_address = { .port = GPIO_PORT_B, .pin = 5 };
+  GpioAddress fan_pin_address = { .port = GPIO_PORT_B, .pin = 3 };
+  bool fan_speed_toggle = FAN_FULL_SPEED;
```
Comment (copperium) [inline projects/smoketest_max6643/src/main.c:31]: Sorry, I meant you don't have to use these intermediary variables between the #defines and the Max6643Settings. You still need the defines, you just don't need to assign them to variables, so the settings struct would have lines like `.overtemp_pin  = OVERTEMP_PIN`.

**Review by copperium — CHANGES_REQUESTED** [review]
One more thing, a bit of a miscommunication.

File: projects/smoketest_max6643/src/main.c (hunk 5 above)
Comment (nourishnew) [reply projects/smoketest_max6643/src/main.c]: done

**Review by copperium — APPROVED** [review]

**Pull request merged**